test/extensions/filters/http/admission_control/config_test.cc

```diff
@@ -75,8 +75,10 @@ sampling_window: 1337s
   TestUtility::loadFromYamlAndValidate(yaml, proto);
   NiceMock<Server::Configuration::MockFactoryContext> factory_context;
   EXPECT_THROW_WITH_MESSAGE(
-      admission_control_filter_factory.createFilterFactoryFromProtoTyped(
-          proto, "whatever", dual_info_, factory_context.getServerFactoryContext()),
+      EXPECT_TRUE(admission_control_filter_factory
+                      .createFilterFactoryFromProtoTyped(proto, "whatever", dual_info_,
+                                                         factory_context.getServerFactoryContext())
+                      .ok()),
```

Comment:
I don't understand how this construct is working? `EXPECT_TRUE` doesn't throw an exception does it? And if the inner part threw an exception then `.ok()` wouldn't be true so it seems like it ought to fail either way?
Even if it does work, I think the concept would be more readable like
```
EXPECT_THAT(admission_control_filter_factory
                      .createFilterFactoryFromProtoTyped(proto, "whatever", dual_info_,
                                                         factory_context.getServerFactoryContext()), 
    HasStatusMessage(HasSubstr("Success rate threshold cannot be less than 1.0%.")));
```
Or, if the thing is that it's throwing an exception and the EXPECT_TRUE is being ignored, then
```
  EXPECT_THROW_WITH_MESSAGE(
      admission_control_filter_factory
                      .createFilterFactoryFromProtoTyped(proto, "whatever", dual_info_,
                                                         factory_context.getServerFactoryContext())
                      .IgnoreResult(),
    EnvoyException, "Success rate threshold cannot be less than 1.0%.");
```

Reply:
the assert is being ignored. 
 no member named 'IgnoreResult' in 'absl::StatusOr<std::function<void (Envoy::Http::FilterChainFactoryCallbacks &)>>' ditto absl::Status  not sure what you're suggesting I use?